SYCL/DeviceLib/sycl_ext_intel_math_fp16.cpp

```diff
@@ -0,0 +1,72 @@
+// RUN: %clangxx -fsycl %s -o %t.out
+// RUN: %CPU_RUN_PLACEHOLDER %t.out
+// RUN: %ACC_RUN_PLACEHOLDER %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+
+// RUN: %clangxx -fsycl -fno-builtin -fsycl-device-lib-jit-link %s -o %t.out
+// RUN: %CPU_RUN_PLACEHOLDER %t.out
+// RUN: %ACC_RUN_PLACEHOLDER %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+//
+// UNSUPPORTED: cuda || hip
+
+#include "imf_utils.hpp"
+#include <sycl/ext/intel/math.hpp>
+
+class is_close {
+public:
+  bool operator()(sycl::half x, sycl::half y) {
+    if (sycl::fabs((x - y) / (x + y)) <
+        std::numeric_limits<sycl::half>::epsilon() * 2)
+      return true;
+    return false;
+  }
+};
+
+class is_close2 {
+public:
+  bool operator()(sycl::half2 x, sycl::half2 y) {
+    if ((sycl::fabs((x.s0() - y.s0()) / (x.s0() + y.s0())) <
+         std::numeric_limits<sycl::half>::epsilon() * 2) &&
+        (sycl::fabs((x.s1() - y.s1()) / (x.s1() + y.s1())) <
+         std::numeric_limits<sycl::half>::epsilon() * 2))
+      return true;
+    return false;
+  }
+};
+
+int main(int, char **) {
+  sycl::queue device_queue(sycl::default_selector_v);
+  std::cout << "Running on "
+            << device_queue.get_device().get_info<sycl::info::device::name>()
+            << "\n";
+
+  if (!device_queue.get_device().has(sycl::aspect::fp16)) {
+    std::cout << "Test skipped on platform without fp16 support." << std::endl;
+    return 0;
+  }
+
+  // sycl::ext::intel::math::inv
+  {
+    std::initializer_list<sycl::half> input_vals = {1.f, 2.0f,  0.5f,  100.f,
+                                                    3.f, 0.25f, 0.002f};
+    std::initializer_list<sycl::half> ref_vals = {
+        1.f, 0.5f, 2.0f, 0.01f, 0.333333333f, 4.f, 500.f};
+    constexpr auto inv_op = [](sycl::half x) {
+      return sycl::ext::intel::math::inv(x);
+    };
+    test<sycl::half, sycl::half, decltype(inv_op), is_close>(
+        device_queue, input_vals, ref_vals, inv_op);
+
+    constexpr auto inv_op2 = [](sycl::half2 x) {
+      return sycl::ext::intel::math::inv(x);
+    };
+    std::initializer_list<sycl::half2> input_vals2 = {
+        {1.f, 2.0f}, {0.5f, 100.f}, {3.f, 0.25f}, {0.002f, 0.005f}};
+    std::initializer_list<sycl::half2> ref_vals2 = {
+        {1.f, 0.5f}, {2.0f, 0.01f}, {0.333333333f, 4.f}, {500.f, 200.f}};
+    test<sycl::half2, sycl::half2, decltype(inv_op2), is_close2>(
+        device_queue, input_vals2, ref_vals2, inv_op2);
+    std::cout << "sycl::ext::intel::math::inv passes" << std::endl;
+  }
```

Comment:
```suggestion
  }
  return 0;
```
Same for the other tests.

Reply:
This comment was only applied to this test, but other tests are missing it too.